File: crop_favicon.py
from PIL import Image
import os

path = r"c:\Users\gmara\OneDrive\Desktop\Directory Dev\Teknovra - Smart Analysis System\src\imports\logo_teknovra.png"
out_path = r"c:\Users\gmara\OneDrive\Desktop\Directory Dev\Teknovra - Smart Analysis System\public\favicon.ico"

img = Image.open(path).convert("RGBA")
bbox = img.getbbox()
if bbox:
    img = img.crop(bbox)

# The logogram is on the left. We assume it's roughly square.
h = img.height
# Crop a square from the left
logogram = img.crop((0, 0, h, h))

logogram.save(out_path, format='ICO', sizes=[(32, 32), (64, 64)])
print("Favicon saved to", out_path)

File: crop_favicon.js
import sharp from 'sharp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function processImage() {
  try {
    const inputPath = path.join(__dirname, 'src/imports/logo_teknovra.png');
    const outputPath = path.join(__dirname, 'public/favicon.png');
    
    // Read the image
    const image = sharp(inputPath);
    const metadata = await image.metadata();
    
    // We assume logogram is on the left. We crop a square box based on height
    // Ensure we don't exceed width
    const size = Math.min(metadata.width, metadata.height);
    
    await image
      .extract({ left: 0, top: 0, width: size, height: size })
      .resize(128, 128)
      .png()
      .toFile(outputPath);
      
    console.log("Successfully created favicon.png", { width: metadata.width, height: metadata.height, croppedSize: size });
  } catch (err) {
    console.error("Error cropping image:", err);
  }
}

processImage();
